### ISSUE TITLE
Determine unobserved Greengenes OTUs

### ISSUE THREAD

**Issue by wasade** [qiime/qiime_web_app#55]
Is it easy to determine what Greengenes OTUs have not been observed? I'd like to check them for chimeras or low quality Sanger calls.

**Comment by douginator2000** [comment]
Here's a basic rundown:
- 408135 total entries in the greengenes_reference table.
- 374378 unobserved OTUs in our database (~92%).
- 33757 observed species. 

This agrees exactly with the distinct count of observed species in our OTU table so hopefully that's a good sanity check.

Given the large size of the unobserved list, I'll post queries to give you full results.

-- Count of total entries in greengenes_reference
select count(*) from greengenes_reference;

-- Count of unobserved OTUs
select  count(distinct gr.prokmsa_id)
from    greengenes_reference gr
        left join otu_table ot
        on gr.prokmsa_id = ot.reference_id
where   ot.reference_id is null;

-- List of unobserved OTUs. You can switch the taxonomy_str filter
-- if PHPR_tax_string is a more desirable output.
select  distinct gr.prokmsa_id, gr.genbank_acc_id, gt.taxonomy_str
from    greengenes_reference gr
        inner join greengenes_taxonomy gt
        on gr.prokmsa_id = gt.prokmsa_id
        left join otu_table ot
        on gr.prokmsa_id = ot.reference_id
where   ot.reference_id is null
        and gt.taxonomy_name = 'ncbi_tax_string'
        -- and gt.taxonomy_name = 'PHPR_tax_string'
order by gt.taxonomy_str;

**Comment by douginator2000** [comment]
(Please close if this satisfies the request. Thanks!)

**Comment by wasade** [comment]
does not. that is a comparison of the full reference set (408k taxa) against the 97% otus. need to ask what percent of the 97% otus are not observed

**Comment by douginator2000** [comment]
Thanks for the clarification Daniel. Updated numbers here:
- Total number of GG 97 OTUs in reference set: 35429
- Total number of unobserved 97% OTUs at 97% similarity: 1672 (4.7%)

-- Total count of GG 97% OTUs 
select  count(*)
from    greengenes_reference gr
        inner join source_map sm
        on gr.prokmsa_id = sm.reference_id
where   sm.sequence_source_id = 4;

-- Total number of unobserved 97% GG OTUs at 97% similarity
select  count(distinct gr.prokmsa_id)
from    greengenes_reference gr
        inner join source_map sm
        on gr.prokmsa_id = sm.reference_id
        left join otu_table ot
        on gr.prokmsa_id = ot.reference_id
where   ot.reference_id is null
        and sm.sequence_source_id = 4;

-- List of unobserved OTUs with taxonomy info
select  count(distinct gr.prokmsa_id, gr.genbank_acc_id, gt.taxonomy_str)
from    greengenes_reference gr
        inner join greengenes_taxonomy gt
        on gr.prokmsa_id = gt.prokmsa_id
        inner join source_map sm
        on gr.prokmsa_id = sm.reference_id
        left join otu_table ot
        on gr.prokmsa_id = ot.reference_id
where   ot.reference_id is null
        and sm.sequence_source_id = 4
        and gt.taxonomy_name = 'ncbi_tax_string'
        -- and gt.taxonomy_name = 'PHPR_tax_string'
order by gt.taxonomy_str;

**Comment by douginator2000** [comment]
A list of the unobserved OTUs:

https://docs.google.com/spreadsheet/ccc?key=0As1evoSSZgradHhsTG1XYXNfSVZBcFJTVi1kY0hKUmc

**Comment by douginator2000** [comment]
Note that this covers (very approximately):
~ Number of sequences 2,750,000,000+
~ Number of samples: 33,000+
~ Biomes represented: 50+

**Comment by rob-knight** [comment]
That makes a lot more sense as the first numbers didn't match our earlier claim that 92% were covered by EMP...thanks for calculating!

On Jan 17, 2013, at 3:09 PM, Doug Wendel <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks for the clarification Daniel. Updated numbers here:
-   Total number of GG 97 OTUs in reference set: 35429
-   Total number of unobserved 97% OTUs at 97% similarity: 1672 (4.7%)

-- Total count of GG 97% OTUs
select count(*)
from greengenes_reference gr
inner join source_map sm
on gr.prokmsa_id = sm.reference_id
where sm.sequence_source_id = 4;

-- Total number of unobserved 97% GG OTUs at 97% similarity
select count(distinct gr.prokmsa_id)
from greengenes_reference gr
inner join source_map sm
on gr.prokmsa_id = sm.reference_id
left join otu_table ot
on gr.prokmsa_id = ot.reference_id
where ot.reference_id is null
and sm.sequence_source_id = 4;

-- List of unobserved OTUs with taxonomy info
select count(distinct gr.prokmsa_id, gr.genbank_acc_id, gt.taxonomy_str)
from greengenes_reference gr
inner join greengenes_taxonomy gt
on gr.prokmsa_id = gt.prokmsa_id
inner join source_map sm
on gr.prokmsa_id = sm.reference_id
left join otu_table ot
on gr.prokmsa_id = ot.reference_id
where ot.reference_id is null
and sm.sequence_source_id = 4
and gt.taxonomy_name = 'ncbi_tax_string'
-- and gt.taxonomy_name = 'PHPR_tax_string'
order by gt.taxonomy_str;

**Comment by rob-knight** [comment]
ok so we are missing
- some obligate pathogens
- some rhizosphere stuff that will be covered by jose's rice project if he ever gets metadata
- some hydrothermal vent organisms (which janet is trying to line up samples from)
- some gut taxa, which seem to have infinite diversity...

pretty good going really!

On Jan 17, 2013, at 3:09 PM, Doug Wendel <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks for the clarification Daniel. Updated numbers here:
-   Total number of GG 97 OTUs in reference set: 35429
-   Total number of unobserved 97% OTUs at 97% similarity: 1672 (4.7%)

-- Total count of GG 97% OTUs
select count(*)
from greengenes_reference gr
inner join source_map sm
on gr.prokmsa_id = sm.reference_id
where sm.sequence_source_id = 4;

-- Total number of unobserved 97% GG OTUs at 97% similarity
select count(distinct gr.prokmsa_id)
from greengenes_reference gr
inner join source_map sm
on gr.prokmsa_id = sm.reference_id
left join otu_table ot
on gr.prokmsa_id = ot.reference_id
where ot.reference_id is null
and sm.sequence_source_id = 4;

-- List of unobserved OTUs with taxonomy info
select count(distinct gr.prokmsa_id, gr.genbank_acc_id, gt.taxonomy_str)
from greengenes_reference gr
inner join greengenes_taxonomy gt
on gr.prokmsa_id = gt.prokmsa_id
inner join source_map sm
on gr.prokmsa_id = sm.reference_id
left join otu_table ot
on gr.prokmsa_id = ot.reference_id
where ot.reference_id is null
and sm.sequence_source_id = 4
and gt.taxonomy_name = 'ncbi_tax_string'
-- and gt.taxonomy_name = 'PHPR_tax_string'
order by gt.taxonomy_str;

**Comment by douginator2000** [comment]
Yeah my brain fart there apologies. Pretty cool to see that we've hit almost everything in there.

**Comment by rob-knight** [comment]
No apology needed (unless you are sufficiently familiar to figure out those functional roles from the species names, in which case well done!)

On Jan 17, 2013, at 3:40 PM, Doug Wendel <notifications@github.com<mailto:notifications@github.com>> wrote:

Yeah my brain fart there apologies. Pretty cool to see that we've hit almost everything in there.

**Comment by wasade** [comment]
Of those, 269 of the OTUs cannot be observed due to the homopolymer filter
on split_libraries.py. split_libraries_fastq.py does not have the
homopolymer check as far as I can tell. Doug, can you verify what split
libraries method is used for loading the Illumina data? I believe fastq,
but want to be sure.

Comparing the remaining 1400 or so, about 1100 are in GG_12_10. It is
possible that the 300 that dropped were due to poor quality or chimeric,
and were caught by the updated quality filtering methods . I can dig deeper
there but am not very motivated to unless there is a strong need to resolve
why they were dropped. I'm going to send on the list of the 1100 to Phil
and see if any of them jump as possible crap sequence. I hope/expect that
most of these are real, but would like to gut as much garbage from
Greengenes as possible
